packages/webdriverio/src/scripts/resq.js

```diff
@@ -3,35 +3,25 @@ export const waitToLoadReact = function waitToLoadReact () {
 }
 
 export const react$ = function react$ (selector, props, state) {
-    let elems = window.resq.resq$$(selector)
+    let elem = window.resq.resq$(selector)
         .byProps(props)
         .byState(state)
 
-    if (elems.length === 0) {
+    if (!elem.name) {
         return { message: `React element with selector "${selector}" wasn't found` }
     }
 
-    let elem = elems[0]
-
-    if (elem.node) {
-        return elem.node
-    }
-
-    elem = elem.node || elem.children[0].node
-    return elem
+    return elem.node || elem.children[0].node
 }
 
 export const react$$ = function react$$ (selector, props, state) {
     let elems = window.resq.resq$$(selector)
         .byProps(props)
         .byState(state)
 
-    return [...elems].map((elem) => {
-        if (elem.node) {
-            return elem.node
-        }
+    if (!elems.length) {
+        return { message: `React element with selector "${selector}" wasn't found` }
```

Comment:
I think will be more precise `React elements...` instead of `element`

Reply:
Adjusted here https://github.com/webdriverio/webdriverio/pull/3819/commits/230caf4b1cc81bcfb5cd9d60aace6466bebb8677#diff-6e003f06aad6f9b928c5d4923fcd06caR52